src/number_theory/cyclotomic/basic.lean

```diff
@@ -335,10 +331,12 @@ begin
 end
 
 /-- If `S` is finite and `is_cyclotomic_extension S A B`, then `B` is a finite `A`-algebra. -/
-lemma finite [is_domain B] [h₁ : fintype S] [h₂ : is_cyclotomic_extension S A B] : finite A B :=
+lemma finite [is_domain B] [h₁ : _root_.finite S] [h₂ : is_cyclotomic_extension S A B] :
```

Comment:
worth `protecting` this so that you don't have to write `_root_` everywhere?

Reply:
It's clashing with `module.finite`, so I've protected it and removed `open module`.

But there is something strange, it is still complaining about ambiguous overload, with `is_cyclotomic_extension.finite`. Maybe `protected` doesn't work in the same file/namespace/whatever where the lemma is declared?